File: singleno.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap <Integer,Integer> map= new HashMap<>();

        for(int i : nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }

        for(Map.Entry<Integer,Integer> entry:map.entrySet()){
            int k = entry.getKey();
            int v = entry.getValue();

            if(v == 1){
                return k;
            }
        }
        return -1;
    }
}

File: singleno.cpp
class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        unordered_map<int, int> map;

        for (int n : nums)
        {
            map[n]++;
        }

        for (auto pair : map)
        {
            int v = pair.first;
            int s = pair.second;
            if (s == 1)
            {
                return v;
            }
        }
        return -1;
    }
};